src/kinetics/Reaction.cpp

```diff
@@ -1039,14 +1036,167 @@ std::string ThreeBodyReaction3::reactantString() const
 
 std::string ThreeBodyReaction3::productString() const
 {
-    if (specified_collision_partner) {
+    if (m_third_body->specified_collision_partner) {
         return ElementaryReaction3::productString() + " + "
             + m_third_body->efficiencies.begin()->first;
     } else {
         return ElementaryReaction3::productString() + " + M";
     }
 }
 
+BlowersMaselReaction::BlowersMaselReaction()
+{
+    setRate(newReactionRate(type()));
+}
+
+BlowersMaselReaction::BlowersMaselReaction(
+        const Composition& reactants, const Composition& products,
+        const BlowersMaselRate& rate)
+    : Reaction(reactants, products)
+{
+    m_rate.reset(new BlowersMaselRate(rate));
+}
+
+BlowersMaselReaction::BlowersMaselReaction(const AnyMap& node, const Kinetics& kin)
+{
+    if (!node.empty()) {
+        setParameters(node, kin);
+        setRate(newReactionRate(node, calculateRateCoeffUnits3(kin)));
+    } else {
+        setRate(newReactionRate(type()));
+    }
+}
+
+FalloffReaction3::FalloffReaction3()
+    : Reaction()
+{
+    m_third_body.reset(new ThirdBody);
+    m_third_body->mass_action = false;
+    setRate(newReactionRate(type()));
+}
+
+FalloffReaction3::FalloffReaction3(const Composition& reactants,
+                                   const Composition& products,
+                                   const ReactionRate& rate,
+                                   const ThirdBody& tbody)
+    : Reaction(reactants, products)
+{
+    m_third_body = std::make_shared<ThirdBody>(tbody);
+    m_third_body->mass_action = false;
+    AnyMap node = rate.parameters();
+    node.applyUnits();
+    std::string rate_type = node["type"].asString();
+    if (rate_type != "falloff" && rate_type != "chemically-activated") {
+        // use node information to determine whether rate is a falloff rate
+        throw CanteraError("FalloffReaction3::FalloffReaction3",
+            "Incompatible types: '{}' is not a falloff rate object.", rate.type());
+    }
+    setRate(newReactionRate(node));
+}
+
+FalloffReaction3::FalloffReaction3(const AnyMap& node, const Kinetics& kin)
+{
+    m_third_body.reset(new ThirdBody);
+    m_third_body->mass_action = false;
+    if (!node.empty()) {
+        setParameters(node, kin);
+        setRate(newReactionRate(node, calculateRateCoeffUnits3(kin)));
+    } else {
+        setRate(newReactionRate(type()));
+    }
+}
+
+std::string FalloffReaction3::type() const
+{
+    if (m_rate && \
+            std::dynamic_pointer_cast<FalloffRate>(m_rate)->chemicallyActivated()) {
+        return "chemically-activated";
+    }
```

Comment:
The backslash line continuation is unnecessary here. I'd suggest formatting this as
```suggestion
    if (m_rate &&
        std::dynamic_pointer_cast<FalloffRate>(m_rate)->chemicallyActivated())
    {
        return "chemically-activated";
    }
```